lib/Feature/FeatureModel.cpp

```diff
@@ -14,28 +14,15 @@ void FeatureModel::dump() const {
   llvm::outs() << '\n';
 }
 
-bool FeatureModel::addFeature(std::unique_ptr<Feature> NewFeature) {
-  // TODO(s9latimm): check consistency
+Feature *FeatureModel::addFeature(std::unique_ptr<Feature> NewFeature) {
   auto FeatureName = std::string(NewFeature->getName());
   if (!Features.try_emplace(FeatureName, std::move(NewFeature)).second) {
-    return false;
+    return nullptr;
   }
   auto *InsertedFeature = Features[FeatureName].get();
```

Comment:
If I remember correctly `try_emplace` returns and iterator, bool pair.
You create a local var with this and then use it to make the check and you don't have to lookup the feature.
```
auto PosInsertedFeature = Features.try_emplace(...)
if (PosInsertedFeature.second) {
  return nullptr;
}
...
```